api/spec/v1.yaml

```diff
@@ -355,6 +357,36 @@ paths:
                 $ref: '#/components/schemas/ConsensusEventList'
         <<: *common_error_responses
 
+  /consensus/roothash_messages:
```

Comment:
Would it make sense to move this endpoint under `/{runtime}/roothash_messages`? It would better reflect the fact that `runtime` is pretty much a required arg (see also thread below), and it would probably align better with how it will be used in Explorer.

I know that the runtime is not where messages live architecturally in oasis-core, but I think the implied expectation is that we'll be showing these on the block page for each _runtime_ block? If we'll be showing these somewhere on the consensus part of the Explorer, do we need the ability to filter this by consensus height? (Specifying consensus_height=H would probably mean "give me roothash messages of runtime rounds that were finalized at H")

Reply:
not sure. maybe, if we go down the route of requiring the 'runtime' filter. and agreed on the weirdness of having a consensus layer thing not in /consensus/.

it would be good to see it on the consensus side because these are so much like consensus transactions. but it'll take some thought on where to put them and how we'll want to query them

Reply:
> these are so much like consensus transactions

Mmm that's a strong point. And the runtime is like the tx sender. So this dilemma is analogous to having `/transactions?sender=X` rather than `/accounts/X/transactions`, and we do do the former. Let's keep it as-is then 👍 .

Reply:
k keeping it here